spyder/plugins/ipythonconsole/widgets/client.py

```diff
@@ -273,12 +273,20 @@ def configure_shellwidget(self, give_focus=True):
         # To apply style
         self.set_color_scheme(self.shellwidget.syntax_style, reset=False)
 
-        # To hide the loading page
-        self.shellwidget.sig_prompt_ready.connect(self._hide_loading_page)
+    def pre_configure_shellwidget(self):

```

Comment:
There's no configuration done to the shellwidget in this method. Therefore, I  think it's better to name it `before_prompt_is_ready`.